x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/Case.java

```diff
@@ -120,8 +120,18 @@ protected TypeResolution resolveType() {
      */
     @Override
     public boolean foldable() {
-        return (conditions.isEmpty() && elseResult.foldable()) ||
-            (conditions.size() == 1 && conditions.get(0).condition().foldable() && conditions.get(0).result().foldable());
+        if (conditions.isEmpty() && elseResult.foldable()) {
+            return true;
+        }
+        if (conditions.size() == 1 && conditions.get(0).condition().foldable()) {
+            // Need to prevent ConstantFolding rule to get applied before SimplifyCase rule
```

Comment:
I find the comment misleading as it implies the rules order is incorrect - called in a different order the rules might affect `fold` but the matter is, `foldable` is incorrect (which leads to the exception).